### PR TITLE
Avoid creating .cache directory in cwd

### PR DESCRIPTION
**Pull request by ssbarnea**
Instead of using cwd for the .cache folder, we now try to use the project_dir for that, which defaults to the location of
.ansiblelint config file.

Running ansible-lint from a subdirectory of the project, should no longer create .cache folders in new locations.

Execution outside a project or a git repository will make the linter use user home directory as reference,

### CODE REVIEW

File: docs/usage.rst [hunk 1]
```diff
@@ -23,6 +23,19 @@ associated tags.
...
+
+As part of the execution, the linter will likely need to create a cache of
+installed or mocked roles, collections and modules. This is done inside
+``{project_dir}/.cache`` folder. The project directory is either given as a
+command line argument, determined by location of the configuration
+file, git project top-level directiory or user home directory as fallback.
+In order to speed-up reruns, the linter does not clean this folder by itself.
+
+If you are using git, you will likely want to add this folder to your
+``.gitignore`` file by running ``git ignore .cache``.
```
Comment (aminvakil) [inline docs/usage.rst:37]:
```suggestion
``.gitignore`` file.
```

**Review by aminvakil — COMMENTED** [review]
I'm not sure what magic git you're running :) but on latest git installation (2.30.1) from https://git-scm.com/ there is no `git ignore` command as far as I know.

Edit: As far as I can tell other than this little documentation fix, everything else is ok.

**Comment by ssbarnea** [comment]
> I'm not sure what magic git you're running :) but on latest git installation (2.30.1) from https://git-scm.com/ there is no `git ignore` command as far as I know.
> 
> Edit: As far as I can tell other than this little documentation fix, everything else is ok.

I do have few git extensions installed, sometimes I endup with mistakes like this. Thanks for reviewing it.

**Review by aminvakil — APPROVED** [review]
Thank you for doing this!

**Pull request merged**